crates/miden-tx/src/executor/exec_host.rs

```diff
@@ -622,64 +623,59 @@ where
-                    self.tx_progress.start_note_execution(clk, note_id);
-                    Ok(Vec::new())
-                },
-                TransactionEvent::NoteExecutionEnd { clk } => {
-                    self.tx_progress.end_note_execution(clk);
-                    Ok(Vec::new())
-                },
-
-                TransactionEvent::TxScriptProcessingStart { clk } => {
-                    self.tx_progress.start_tx_script_processing(clk);
-                    Ok(Vec::new())
-                },
-                TransactionEvent::TxScriptProcessingEnd { clk } => {
-                    self.tx_progress.end_tx_script_processing(clk);
-                    Ok(Vec::new())
-                },
-
-                TransactionEvent::EpilogueStart { clk } => {
-                    self.tx_progress.start_epilogue(clk);
-                    Ok(Vec::new())
-                },
-                TransactionEvent::EpilogueEnd { clk } => {
-                    self.tx_progress.end_epilogue(clk);
-                    Ok(Vec::new())
-                },
-
-                TransactionEvent::EpilogueAuthProcStart { clk } => {
-                    self.tx_progress.start_auth_procedure(clk);
-                    Ok(Vec::new())
-                },
-                TransactionEvent::EpilogueAuthProcEnd { clk } => {
-                    self.tx_progress.end_auth_procedure(clk);
-                    Ok(Vec::new())
-                },
-
-                TransactionEvent::EpilogueAfterTxCyclesObtained { clk } => {
-                    self.tx_progress.epilogue_after_tx_cycles_obtained(clk);
-                    Ok(Vec::new())
+                TransactionEvent::Progress(tx_progress) => match tx_progress {
+                    TransactionProgressEvent::NotesProcessingStart(clk) => {
+                        self.tx_progress.start_notes_processing(clk);
+                        Ok(Vec::new())
+                    },
+                    TransactionProgressEvent::NotesProcessingEnd(clk) => {
+                        self.tx_progress.end_notes_processing(clk);
+                        Ok(Vec::new())
+                    },
+                    TransactionProgressEvent::NoteExecutionStart { note_id, clk } => {
+                        self.tx_progress.start_note_execution(clk, note_id);
+                        Ok(Vec::new())
+                    },
+                    TransactionProgressEvent::NoteExecutionEnd(clk) => {
+                        self.tx_progress.end_note_execution(clk);
+                        Ok(Vec::new())
+                    },
+                    TransactionProgressEvent::PrologueStart(clk) => {
+                        self.tx_progress.start_prologue(clk);
+                        Ok(Vec::new())
+                    },
+                    TransactionProgressEvent::PrologueEnd(clk) => {
+                        self.tx_progress.end_prologue(clk);
+                        Ok(Vec::new())
+                    },
```

Comment:
Nit: Let's rearrange this to handle the events in the same order as they appear in `TransactionProgressEvent`, e.g. first PrologueStart/PrologueEnd, then NotesProcessingStart/NotesProcessingEnd, etc.